packages/interface/src/components/preferences-modal-tabs/style.scss

```diff
@@ -0,0 +1,27 @@
+$vertical-tabs-width: 160px;
+
+.interface-preferences__tabs {
+	.components-tab-panel__tabs {
```

Comment:
Ideally we should try to avoid using internal components classnames (as they are not regarded as "public APIs") and to override styles this way, as those styles can easily break in case the component is updated in the future.

I understand this may also be caused by a limitation in `TabPanel`'s design, or by its misalignment with the recent UI direction — cc'ing @jasmussen here who may have more insights here

Reply:
Thanks for your feedback @ciampo !

These styles were copied verbatim from the original `edit-post` modal, so they've been around for a while now. For `components-tab-panel__tabs` and `components-modal__content` I could potentially target them by aria-role instead - would this be preferable?

The problem is `components-base-control__help` is a `p` tag with no distinctive features, so there's no other way to target it. (By the way: that's an accessibility issue, because the contents of that tag are ignored by screen readers. It should be linked to the control through an `aria-describedby` or similar. I'm happy to create an issue for that if there isn't one already 🙂 )

In more general terms, are there any thoughts around expanding the components API to allow applying styles to child elements of a component? It's quite frequent across the codebase that we resort to the internal classnames for this end. It would be good to find a solution, e.g.:

*  make some of those classnames an official part of the API;
*  provide a way to pass classnames for child elements (this can become messy though)
* provide a CSS custom property API whereby consumers can set certain values for the child elements (colors, spacing etc.) of a component.

Let me know what you think!

Reply:
> These styles were copied verbatim from the original edit-post modal, so they've been around for a while now.

Thank you for the explanation, I previously had missed this detail (my fault, I skipped the PR's description and went straight to the ping from Riad). I'd say, let's try to see if there's any low-hanging fruit (like [this comment](https://github.com/WordPress/gutenberg/pull/39153#r818466474)) and leave more complicated refactors for another time!

> For `components-tab-panel__tabs` and `components-modal__content` I could potentially target them by aria-role instead - would this be preferable?

That would be slightly better, but we can probably leave it as-is in this PR. I would be personally question why we need to override so many styles for the `Modal` and `TabPanel` components — the better solution in my opinion would be to avoid these styles overrides as much as possible. In a follow-up PR we could look at what limitations we're facing when using these components, and see how we can overcome them (by either improving these components or using a different set of components).

> The problem is components-base-control__help is a p tag with no distinctive features, so there's no other way to target it. (By the way: that's an accessibility issue, because the contents of that tag are ignored by screen readers. It should be linked to the control through an aria-describedby or similar. I'm happy to create an issue for that if there isn't one already 🙂 )

Understood — feel free to create an issue for that, and tag @mirka and me.

> In more general terms, are there any thoughts around expanding the components API to allow applying styles to child elements of a component? It's quite frequent across the codebase that we resort to the internal classnames for this end. It would be good to find a solution, e.g.:
>
> *  make some of those classnames an official part of the API;
> *  provide a way to pass classnames for child elements (this can become messy though)
> * provide a CSS custom property API whereby consumers can set certain values for the child elements (colors, spacing etc.) of a component.

On this topic, we think that hardcoded classnames and internal DOM structure should not be part of the public APIs of components, as they would severely limit the amount of internal (private) changes that can be made to the components; most internal changes would likely cause breakages for the consumers of such components.

I also agree that exposing more props for children classnames is not the correct solution, as it would scale very poorly.

Regarding CSS custom properties, it's something that we're going to take a look at soon when we'll start actively working on introducing a lightweight theming API layer.

Another approach that we could have is to try and be more "modular" with our components — e.g. expose a `BaseControl` and a `BaseControlHelp` components that the consumer of the library would need to use together (rather than exporting one monolithic `BaseControl` component).

cc @mirka

Reply:
I've created #39271 to track the issue with the help text. I also fixed the CSS targeting `components-navigator-provider` so I think that's all your feedback addressed. Thanks @ciampo !